### PR TITLE
Added TraceLogger to get backtrace for executed queries

### PR DESCRIPTION
**Pull request by pierredup**
Added a TraceLogger that returns a backtrace for the executed query.
The backtrace returns the last 10 frames, and it also formats the frames for easy reading

### CODE REVIEW

File: lib/Doctrine/DBAL/Logging/TraceLogger.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        if ($this->enabled) {
+            $this->queries[$this->currentQuery]['executionMS'] = microtime(true) - $this->start;
+        }
+    }
+
+    private function getBactrace()
+    {
+        $backtrace = debug_backtrace(DEBUG_BACKTRACE_PROVIDE_OBJECT | DEBUG_BACKTRACE_IGNORE_ARGS);
+
+        foreach ($backtrace as $key => $debug) {
```
Comment (deeky666) [inline lib/Doctrine/DBAL/Logging/TraceLogger.php]: So is there any conclusion on this? How should we proceed?

[the remark this answers is not in the document]
Comment (greg0ire) [reply lib/Doctrine/DBAL/Logging/TraceLogger.php]: How about making it optional and configurable?

File: lib/Doctrine/DBAL/Logging/TraceLogger.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+     * {@inheritdoc}
+     */
+    public function stopQuery()
+    {
+        if ($this->enabled) {
+            $this->queries[$this->currentQuery]['executionMS'] = microtime(true) - $this->start;
+        }
+    }
+
+    private function getBactrace()
```
Comment (greg0ire) [inline lib/Doctrine/DBAL/Logging/TraceLogger.php]: Should be "Backtrace" with a "k"

File: lib/Doctrine/DBAL/Logging/TraceLogger.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pierstoval) [reply lib/Doctrine/DBAL/Logging/TraceLogger.php]: 👍 for optional and configurable with a simple boolean, and I'd make it `true` by default to satisfy everyone

[the remark this answers is not in the document]
Comment (pierredup) [reply lib/Doctrine/DBAL/Logging/TraceLogger.php]: @greg0ire Can you please explain which part you mean to make optional and configurable? This logger is already enabled by default, and can be disabled by setting the property `enabled` to `false`. Is there another part that you want configurable?

[the remark this answers is not in the document]
Comment (greg0ire) [reply lib/Doctrine/DBAL/Logging/TraceLogger.php]: The part I mean is the part where you include the backtrace in the `$queries` array.

**Comment by morozov** [comment]
No more loggers, please. A PSR-3 adapter would be much more welcome.

**Comment by lcobucci** [comment]
Closing this as per @morozov's comment.

**Comment by ostrolucky** [comment]
Implemented in https://github.com/doctrine/DoctrineBundle/pull/954

**Pull request closed without merging**